docs/book/features/step-fixtures.md

```diff
@@ -0,0 +1,97 @@
+there are some special parameters that can be passed into a step function, which serve different needs.
+
+* A `StepContext` object: This object gives access to the artifact store, metadata store, materializers, and special 
+integration-specific libraries.
+* An object which is a subclass of `BaseStepConfig`: This object is used to pass run-time parameters to a pipeline run. It can 
+be used to send parameters to a step that are not artifacts.
+
+These special parameters are similar to [pytest fixtures](https://docs.pytest.org/en/6.2.x/fixture.html), and so we borrow that 
+nomenclature for ZenML: We call them `Step Fixtures`. 
+
+`Step Fixtures` are simple to use: Simply pass a parameter in with the right type hint as follows:
+
+## Using fixtures in Functional API
+
+```python
+@step
+def my_step(
+    context: StepContext,
+    config: SubClassBaseStepConfig,  # subclass of `BaseStepConfig`
+    artifact: str,  # all other parameters are artifacts, coming from upstream steps
+):
+    ...
+```
+
+## Use fixtures in Class-based API
+
+```python
+class MyStep(BaseStep):
+    def entrypoint(
+        self,
+        context: StepContext,
+        config: SubClassBaseStepConfig,  # subclass of `BaseStepConfig`
+        artifact: str,  # all other parameters are artifacts, coming from upstream steps
+    ):
+        ...
+```
+
+Please note in both examples above that the name of the parameter can be anything, but the typehint is what is important.
+
+
+
+## Using the `StepContext`
+
+`StepContext` provides additional context inside a step function.  It is used to access materializers and artifact URIs inside a step function. 
+
+You do not need to create a StepContext object yourself and pass it when creating the step, as long as you specify 
+it in the signature ZenML will create the `StepContext` and automatically pass it when executing your step.
+
+Note: When using a StepContext inside a step, ZenML disables caching for this step by default as the context provides 
+access to external resources which might influence the result of your step execution. 
+To enable caching anyway, explicitly enable it in the @step decorator or when initializing your custom step class.
+
+Within a step, there are many things that you can use the `StepContext` object for. E.g.: 
+
+```python
+@enable_INTEGRATION  # can be `enable_whylogs`, `enable_mlflow` etc. 
+@step
+def my_step(
+    context: StepContext,
+):
+    context.get_output_materializer()  # Returns a materializer for a given step output.
+    context.get_output_artifact_uri()  # Returns the URI for a given step output.
+    context.metadata_store  # Access to the [Metadata Store](https://apidocs.zenml.io/latest/api_docs/metadata_stores/)
+    context.INTEGRATION  # Access to an integration, e.g. `context.whylogs`
+```
+
+For more information, check the [API reference](https://apidocs.zenml.io/latest/api_docs/steps/)
+
+## Using the `BaseStepConfig`
```

Comment:
Not sure if that is something that we should mention in this exact place, but there is one key difference between the `BaseStepConfig` and the `StepContext`. The `StepContext` can never be passed manually by the user and is therefore treated like an actual pytest fixture, but config instances can actually be passed when creating a step. Additionally, just using the config as a fixture without passing an instance only works if every attribute has a default value or gets sets using a config file.

Reply:
Resolved!